doc/neps/nep-0035-array-creation-dispatch-with-array-function.rst

```diff
@@ -317,6 +329,81 @@ downsides pointed out above we have decided to discard any changes on the Python
 side and resolve those issues with a pure-C implementation. Please refer to
 [implementation]_ for details.
 
+Reading the Reference Array Downstream
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+As stated in the beginning of :ref:`neps.like-kwarg.implementation` section,
+``like=`` is not propagated to the downstream library, nevertheless, it's still
+possible to access it. This requires some changes in the downstream library's
+``__array_function__`` definition, where the ``self`` attribute is in practice
+that passed via ``like=``. This is the case because we use ``like=`` as the
+dispatching array, unlike other compute functions covered by NEP-18 that usually
+dispatch on the first positional argument.
+
+An example of such use is to create a new Dask array while preserving its
+backend type:
+
+.. code:: python
+    # Returns dask.array<array, shape=(3,), dtype=int64, chunksize=(3,), chunktype=cupy.ndarray>
+    np.asarray([1, 2, 3], like=da.array(cp.array(())))
+
+    # Returns a cupy.ndarray
+    type(np.asarray([1, 2, 3], like=da.array(cp.array(()))).compute())
```

Comment:
Might be nice to paste the output so that the user can read up that the next sentence is true, but just a thought and a tiny one.

Reply:
I thought about this, but all the code here is "copy-pasteable", thus I added the output as comments instead. Therefore, I'd rather not add the outputs risking making it more confusing.